File: dashboard/src/interfaces/team.ts
export interface Team {
  id: string;
  name: string;
  avatarUrl?: string;
  preferredRegion?: string;
  ownerId: string;
  createdAt: string;
  updatedAt: string;
}

export interface TeamMember {
  id: string;
  teamId: string;
  userId: string;
  userEmail: string;
  role: string;
  joinedAt: string;
}

export interface TeamInvite {
  id: string;
  teamId: string;
  email: string;
  role: string;
  token: string;
  invitedBy: string;
  expiresAt: string;
  createdAt: string;
}

export interface CreateTeamRequest {
  name: string;
  avatarUrl?: string;
  preferredRegion?: string;
}

export interface InviteMemberRequest {
  email: string;
  role: string;
}

export interface AcceptInviteRequest {
  token: string;
}

export interface GetTeamResponse {
  team: Team;
  members: TeamMember[];
}

export interface Workspace {
  id: string;
  name: string;
  avatarUrl?: string;
  preferredRegion?: string;
  ownerId: string;
  createdAt: string;
  updatedAt: string;
}

export interface TrustedDomain {
  id: string;
  teamId: string;
  domain: string;
  role: string;
  createdAt: string;
}

export interface SSHKey {
  id: string;
  teamId: string;
  name: string;
  publicKey: string;
  createdAt: string;
}

export interface AuditLog {
  id: string;
  teamId: string;
  userId?: string;
  projectId?: string;
  environmentId?: string;
  action: string;
  resource?: string;
  actor: string;
  ipAddress?: string;
  createdAt: string;
  timestamp?: string;
}

export interface CreateWorkspaceRequest {
  name: string;
  avatarUrl?: string;
  preferredRegion?: string;
}

export interface UpdateWorkspaceRequest {
  name?: string;
  avatarUrl?: string;
  preferredRegion?: string;
}

export interface CreateTrustedDomainRequest {
  domain: string;
  role: string;
}

export interface CreateSSHKeyRequest {
  name: string;
  publicKey: string;
}

File: internal/cloud/models/team.go
package models

import "time"

type Team struct {
	ID              string    `json:"id"`
	Name            string    `json:"name"`
	AvatarURL       string    `json:"avatarUrl,omitempty"`
	PreferredRegion string    `json:"preferredRegion,omitempty"`
	OwnerID         string    `json:"ownerId"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

type TeamMember struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"teamId"`
	UserID    string    `json:"userId"`
	UserEmail string    `json:"userEmail"`
	Role      string    `json:"role"`
	JoinedAt  time.Time `json:"joinedAt"`
}

type TeamInvite struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"teamId"`
	Email     string    `json:"email"`
	Role      string    `json:"role"`
	Token     string    `json:"token"`
	InvitedBy string    `json:"invitedBy"`
	ExpiresAt time.Time `json:"expiresAt"`
	CreatedAt time.Time `json:"createdAt"`
}

type CreateTeamRequest struct {
	Name            string `json:"name"`
	AvatarURL       string `json:"avatarUrl,omitempty"`
	PreferredRegion string `json:"preferredRegion,omitempty"`
}

type InviteMemberRequest struct {
	Email string `json:"email"`
	Role  string `json:"role"`
}

type AcceptInviteRequest struct {
	Token string `json:"token"`
}

type GetTeamResponse struct {
	Team    *Team         `json:"team"`
	Members []*TeamMember `json:"members"`
}

type Workspace struct {
	ID              string    `json:"id"`
	Name            string    `json:"name"`
	AvatarURL       string    `json:"avatarUrl,omitempty"`
	PreferredRegion string    `json:"preferredRegion,omitempty"`
	OwnerID         string    `json:"ownerId"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

type TrustedDomain struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"teamId"`
	Domain    string    `json:"domain"`
	Role      string    `json:"role"`
	CreatedAt time.Time `json:"createdAt"`
}

type SSHKey struct {
	ID        string    `json:"id"`
	TeamID    string    `json:"teamId"`
	Name      string    `json:"name"`
	PublicKey string    `json:"publicKey"`
	CreatedAt time.Time `json:"createdAt"`
}

type AuditLog struct {
	ID            string    `json:"id" gorm:"primarykey"`
	TeamID        string    `json:"teamId"`
	UserID        string    `json:"userId,omitempty"`
	ProjectID     string    `json:"projectId,omitempty"`
	EnvironmentID string    `json:"environmentId,omitempty"`
	Action        string    `json:"action"`
	Resource      string    `json:"resource,omitempty"`
	Actor         string    `json:"actor"`
	IPAddress     string    `json:"ipAddress,omitempty"`
	CreatedAt     time.Time `json:"createdAt"`
	Timestamp     time.Time `json:"timestamp,omitempty"`
}

type CreateWorkspaceRequest struct {
	Name            string `json:"name"`
	AvatarURL       string `json:"avatarUrl,omitempty"`
	PreferredRegion string `json:"preferredRegion,omitempty"`
}

type UpdateWorkspaceRequest struct {
	Name            string `json:"name,omitempty"`
	AvatarURL       string `json:"avatarUrl,omitempty"`
	PreferredRegion string `json:"preferredRegion,omitempty"`
}

type CreateTrustedDomainRequest struct {
	Domain string `json:"domain"`
	Role   string `json:"role"`
}

type CreateSSHKeyRequest struct {
	Name      string `json:"name"`
	PublicKey string `json:"publicKey"`
}
